add Google Season of Docs project proposal page (#19215)
cilium-team/cilium: Documentation/contributing/development/season_of_docs.rst

.. only:: not (epub or latex or html)

    WARNING: You are looking at unreleased Cilium documentation.
    Please use the official rendered version released here:
    https://docs.cilium.io

.. _season_of_docs:

Google Season of Docs 2022
==========================

Cilium is interested in applying to the 2022 Google Season of Docs to improve the documentation experience for our users and contributors.
Below is our project ideas, if you have any questions, please reach out to gsod@cilium.io.

Reorganize Cilium Features and Getting Started 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

About Our Organization
~~~~~~~~~~~~~~~~~~~~~~

> Cilium (current version 1.12, first released in 2017) is an Apache 2.0-licensed project for providing, 

Reviewer: nit: For what it's worth, 1.12 is the current development version, 1.11 is the latest GA release at this time. We plan to release 1.12 during the program, possibly also 1.13 but we have not yet planned out anything for that version.

Author: thanks updated it to 1.11

> Cilium (current version 1.12, first released in 2017) is an Apache 2.0-licensed project for providing, 

Reviewer 2: ```suggestion
Cilium (current version 1.11, first released in 2017) is an Apache 2.0-licensed project for providing, 
```

securing, and observing network connectivity between container workloads - cloud native, and built around 
the revolutionary Linux Kernel technology eBPF. Our users adopt Cilium to solve their cloud networking, 
observability, and security issues at scale including running some of the largest Kubernetes clusters in the world. 
Cilium has a large community of contributors from over 50 countries around the world. Multiple companies 
have staff paid to work on Cilium too. Cilium is an incubating project in the Cloud Native Computing Foundation 

> and is the networking dataplane for Google Kubernetes Engine and Anthos.

Author: ```suggestion
and is the networking data plane for Google Kubernetes Engine and Anthos.
```

About the Problem
~~~~~~~~~~~~~~~~~

Cloud native networking and security are complex topics that even long time practitioners struggle to master.
Cilium documentation has grown organically over time and the project has mainly been used by advanced cloud native users. 
The documentation lacks the structure that new, and even existing, users can rely on to quickly find the information they need. 
In addition, users sometimes struggle to discover all of the features and functionality of Cilium because the documentation 

> doesn’t make them easily discoverable. Finally, new users often have questions about how to troubleshoot setting up their network.

Author: ```suggestion
does not make them easily discoverable. Finally, new users often have questions about how to troubleshoot setting up their network.
```

#. Simplify the “Getting Started” section by moving advanced and detailed config guides out to more appropriate sections
#. Work with the release team to restructure the documentation and create a process for keeping the documentation up to date going forward

Work that is out of scope:

#. Writing tutorials or reference documentation

The writer will be working closely with Liz Rice, Bill Mulligan, and other Cilium maintainers. 
We estimate the project will take five months to complete.

Measuring Success
~~~~~~~~~~~~~~~~~

Cilium currently receives issues covered in the documentation in both the Slack and through Github issues. It also receives few new 
contributors to the documentation. We believe that this restructuring will result in fewer questions and issues already covered in 
the documentation and more contributors to the documentation since they will know where to add it.

We will track three metrics (number of questions covered in documentation in Slack, number of questions covered in documentation 
in Github issues, and number of pull requests from new contributors) monthly after the documentation is published.

> We would consider the project successful if, after publication of the new documentation each of the following happens:

Reviewer: I presume that these success criteria are something like monthly rates at the beginning & end of the project? Do they account for existing trends?

Reviewer 2: Yes, we'll need to make some estimate of this before the project starts (but we can do that after we've made the application to GSoD)

- The number of Slack questions covered in the documentation decreases by 10%
- The number of Github issues covered in the documentation decreases by 20%
- The number of pull requests from new contributors increases by 15%

Timeline
~~~~~~~~

The project itself will take approximately six months to complete. Once the tech writer is hired, we'll spend a month on 

> tech writer orientation, then move onto the audit and inventory, and spend the last two months focusing on restructuring the documentation.

Reviewer 2: ```suggestion
tech writer orientation, then move onto the audit and inventory, and spend the last three months focusing on restructuring the documentation.
```
